pkg/httpd/webfolder.go

```diff
@@ -20,56 +20,67 @@ func (h *webFolder) Name() string {
 }
 
 func (h *webFolder) CheckValidation() error {
-	apiClient := h.ws.apiClient
-	user := h.ws.CurrentUser()
-	terminalCfg, err := h.ws.apiClient.GetTerminalConfig()
+	if volume, err := SftpCheckValidation(h.ws); err != nil {
+		return err
+	} else {
+		h.volume = volume
+	}
+	return nil
+}
+
+func (h *webFolder) HandleMessage(*Message) {
+}
+
+func (h *webFolder) CleanUp() {
+	close(h.done)
+	h.volume.Close()
+}
+
+func (h *webFolder) GetVolume() *UserVolume {
+	select {
+	case <-h.done:
+		return nil
+	default:
+		return h.volume
+	}
+}
+
+func SftpCheckValidation(ws *UserWebsocket) (*UserVolume, error) {
+	apiClient := ws.apiClient
+	user := ws.CurrentUser()
+	terminalCfg, err := ws.apiClient.GetTerminalConfig()
+
+	uv := &UserVolume{}
 	if err != nil {
 		logger.Errorf("Get terminal config failed: %s", err)
-		return err
+		return uv, err
 	}
 	volOpts := make([]VolumeOption, 0, 5)
 	volOpts = append(volOpts, WithUser(user))
-	volOpts = append(volOpts, WithAddr(h.ws.ClientIP()))
+	volOpts = append(volOpts, WithAddr(ws.ClientIP()))
 	volOpts = append(volOpts, WithTerminalCfg(&terminalCfg))
-	params := h.ws.wsParams
+	params := ws.wsParams
 	targetId := params.TargetId
 	assetId := params.AssetId
 	if assetId == "" {
 		assetId = targetId
 	}
-	if h.ws.ConnectToken != nil {
-		connectToken := h.ws.ConnectToken
+	if ws.ConnectToken != nil {
+		connectToken := ws.ConnectToken
 		volOpts = append(volOpts, WithConnectToken(connectToken))
 	} else {
 		if common.ValidUUIDString(assetId) {
 			assets, err := apiClient.GetUserAssetByID(user.ID, assetId)
 			if err != nil {
 				logger.Errorf("Get user asset %s error: %s", assetId, err)
-				return ErrAssetIdInvalid
+				return uv, ErrAssetIdInvalid
 			}
 			if len(assets) != 1 {
 				logger.Errorf("Get user more than one asset %s: choose first", targetId)
 			}
 			volOpts = append(volOpts, WithAsset(&assets[0]))
 		}
 	}
-	h.volume = NewUserVolume(apiClient, volOpts...)
-	return nil
-}
 
-func (h *webFolder) HandleMessage(*Message) {
-}
-
-func (h *webFolder) CleanUp() {
-	close(h.done)
-	h.volume.Close()
-}
-
-func (h *webFolder) GetVolume() *UserVolume {
-	select {
-	case <-h.done:
-		return nil
-	default:
-		return h.volume
-	}
+	return NewUserVolume(apiClient, volOpts...), nil
 }
```

Comment:
There's no obvious issue found with the provided Go code. However, consider removing unnecessary comments that do not contribute to understanding or making minor formatting improvements like adding spacing around operators and parentheses where appropriate.

The current difference between this code snippet and previous versions is minor; it's important to maintain consistency while updating the documentation or future revisions in case they need to revert changes.

As of 2023 and onwards versioning may change based on ongoing updates from the developer community so I cannot accurately say how these files would be categorized now compared to past years/versions but generally speaking there have been some notable advancements in software development since 2021 (as per my most recent available knowledge). For an optimized version please refer back to their latest release notes which should contain all relevant changes made in each update cycle.

For reference, here’s a link to the GitHub repository if needed:

https://github.com/[GitHub Organization]/[Repository name]